pkg/proxy/iptables/proxier.go

```diff
@@ -960,32 +960,57 @@ func (proxier *Proxier) syncProxyRules() {
 		if svcInfo.NodePort != 0 {
 			// Hold the local port open so no other process can open it
 			// (because the socket might open but it would never work).
-			lp := utilproxy.LocalPort{
-				Description: "nodePort for " + svcNameString,
-				IP:          "",
-				Port:        svcInfo.NodePort,
-				Protocol:    protocol,
+			addresses, err := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+			if err != nil {
+				glog.Errorf("Failed to get node ip address matching nodeport cidr: %v", err)
+				continue
 			}
-			if proxier.portsMap[lp] != nil {
-				glog.V(4).Infof("Port %s was open before and is still needed", lp.String())
-				replacementPortsMap[lp] = proxier.portsMap[lp]
-			} else {
-				socket, err := proxier.portMapper.OpenLocalPort(&lp)
-				if err != nil {
-					glog.Errorf("can't open %s, skipping this nodePort: %v", lp.String(), err)
-					continue
+
+			lps := make([]utilproxy.LocalPort, 0)
```

Comment:
Can I suggest rewriting this in this way:

```
lps := make([]utilproxy.LocalPort, 0)
for address := range addresses {
	lp := utilproxy.LocalPort{`
		Description: "nodePort for " + svcNameString,
		IP: address,
		Port: svcInfo.NodePort,
		Protocol: protocol,
	}
	if utilproxy.IsZeroCIDR(address) {
		lp.IP = ""
                lps.append(lps, lp)
		// If we encounter a zero CIDR, then there is no point in processing the rest of the addresses.	
		break
	} 
        lps.append(lps, lp)
}
```

Saves you 10 lines if you do it here and in the ipvs code :)

Reply:
Definitely make sense! It's fixed now. PTAL.

THANKS!